examples/plot_minimal_pydeseq2_pipeline.py

```diff
@@ -0,0 +1,151 @@
+"""
+PyDESeq 2 pipeline
+===========================
+
+This notebook gives a minimalistic example of how to perform DEA using PyDESeq2.
+
+It allows you to run the PyDESeq2 pipeline either on synthetic data or on one of
+the following TCGA datasets:
+
+- TCGA-BRCA
+- TCGA-COAD
+- TCGA-LUAD
+- TCGA-LUSC
+- TCGA-PAAD
+- TCGA-PRAD
+- TCGA-READ
+- TCGA-SKCM.
+
+While the synthetic data is part of this repository, TCGA data should be
+downloaded as per the instructions in `datasets/`.
+
+Running this pipeline may take a few minutes (~5-10 min for TCGA) depending on
+your setup and on the chosen dataset.
+"""
+
+import os
+import pickle as pkl
+
+from pydeseq2.DeseqDataSet import DeseqDataSet
+from pydeseq2.DeseqStats import DeseqStats
+from pydeseq2.utils import load_example_data
+
+SAVE = False  # whether to save the outputs of this notebook
+
+# %%
+# Data loading
+# ------------
+#
+# See the `datasets` readme for the required data organization.
+#
+
+DATASET = "synthetic"  # or 'TCGA-BRCA', 'TCGA-COAD', etc.
+
+OUTPUT_PATH = f"../output_files/{DATASET}"
+os.makedirs(OUTPUT_PATH, exist_ok=True)  # Create path if it doesn't exist
+
+counts_df = load_example_data(
+    modality="raw_counts",
+    dataset=DATASET,
+    debug=False,
+)
+
+clinical_df = load_example_data(
+    modality="clinical",
+    dataset=DATASET,
+    debug=False,
+)
+
+print(counts_df)
+
+# %%
+# Remove samples for which `high_grade` is NaN.
+
+if DATASET != "synthetic":
+    samples_to_keep = ~clinical_df.high_grade.isna()
+    samples_to_keep.sum()
+    counts_df = counts_df.loc[samples_to_keep]
+    clinical_df = clinical_df.loc[samples_to_keep]
+
+# %%
+# Filter out genes that have less than 10 counts in total
+
+genes_to_keep = counts_df.columns[counts_df.sum(axis=0) >= 10]
+len(genes_to_keep)
+
+counts_df = counts_df[genes_to_keep]
+
```

Comment:
(doesn't apply to synthetic data)

```suggestion
```

Reply:
This should not be removed by explained I believe. The purpose of this example is not to show how to work on the synthetic data but how to get started with deseq2 on any data (and it just happens that we use the synthetic data for it)

Reply:
Ok, in this case we would keep the gene-filtering part:

```suggestion
# %%
# Filter out genes that have less than 10 counts in total

genes_to_keep = counts_df.columns[counts_df.sum(axis=0) >= 10]
len(genes_to_keep)

counts_df = counts_df[genes_to_keep]

```